sources/ClangSharp/Interop/clangsharp.cs

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Microsoft and Contributors. All rights reserved. Licensed under the University of Illinois/NCSA Open Source License. See LICENSE.txt in the project root for license information.
+
+// Ported from https://github.com/llvm/llvm-project/tree/llvmorg-9.0.0/clang/include/clang-c
+// Original source is Copyright (c) the LLVM Project and Contributors. Licensed under the Apache License v2.0 with LLVM Exceptions. See NOTICE.txt in the project root for license information.
+
+using System;
+using System.Runtime.InteropServices;
+
+namespace ClangSharp.Interop
+{
+    public static unsafe partial class clangsharp
+    {
+        private const string libraryPath = "libClangSharp";
+
+        [DllImport(libraryPath, EntryPoint = "clangsharp_Cursor_getBinaryOpcode", CallingConvention = CallingConvention.Cdecl)]
+        public static extern CX_BinaryOperatorKind Cursor_getBinaryOpcode(CXCursor C);
+
+        [DllImport(libraryPath, EntryPoint = "clangsharp_Cursor_getBinaryOpcodeStr", CallingConvention = CallingConvention.Cdecl)]
+        public static extern CXString Cursor_getBinaryOpcodeStr(CX_BinaryOperatorKind Op);
+
+        [DllImport(libraryPath, EntryPoint = "clangsharp_Cursor_getUnaryOpcode", CallingConvention = CallingConvention.Cdecl)]
+        public static extern CX_UnaryOperatorKind Cursor_getUnaryOpcode(CXCursor C);
+
+        [DllImport(libraryPath, EntryPoint = "clangsharp_Cursor_getUnaryOpcodeStr", CallingConvention = CallingConvention.Cdecl)]
+        public static extern CXString Cursor_getUnaryOpcodeStr(CX_UnaryOperatorKind Op);
+
+        [DllImport(libraryPath, EntryPoint = "clangsharp_getCursorExtent", CallingConvention = CallingConvention.Cdecl)]
+        public static extern CXSourceRange getCursorExtent(CXCursor param0);
+
+        [DllImport(libraryPath, EntryPoint = "clangsharp_getNullRange", CallingConvention = CallingConvention.Cdecl)]
+        public static extern CXSourceRange getNullRange();
+
+        [DllImport(libraryPath, EntryPoint = "clangsharp_getRange", CallingConvention = CallingConvention.Cdecl)]
+        public static extern CXSourceRange getRange(CXSourceLocation begin, CXSourceLocation end);
+
+        [DllImport(libraryPath, EntryPoint = "clangsharp_getSpellingLocation", CallingConvention = CallingConvention.Cdecl)]
+        public static extern void getSpellingLocation(CXSourceLocation location, [NativeTypeName("CXFile *")] void** file, [NativeTypeName("unsigned int *")] uint* line, [NativeTypeName("unsigned int *")] uint* column, [NativeTypeName("unsigned int *")] uint* offset);
```

Comment:
The `libClang` API was actually doing `getFileLocation`. The API, when added, was correctly getting the spelling location; but then at some point later there were CI failures and someone changed it to be functionally `getFileLocation` with a `FIXME` comment.

This version of the API correctly gets the spelling location, which allows users to get the line/column/offset as it exists in the raw header files. This is useful for when processing macros.